### PR TITLE
Fixes to the SH-Macro-Parser branch

### PR DESCRIPTION
**Pull request by scunz**
Contains:
- First commit sets the expected result right
- Second and Third commit fix two bugs that show up as "endless recursion", but are actually no recursion at all :-)
- Thrid removes an odd advance that nobody will ever care about
- And the final commit fixes some white space that Qt Creator automatically fixed up :)

By the way: The "artificial else-if construct" that you're using is hard to follow :) Actually, I know some people say the contrary. Just an opinion, though...

### CODE REVIEW

**Comment by antis81** [comment]
@scunz I push these directly to my branch if you don't mind - thanks :+1:. I was thinking too complicated :smile:

Btw. Shouldn't we recures in on the `$` instead of `{`?

**Comment by scunz** [comment]
`$foo` is different from `${foo}`, so we should not reset the recursion before we see the `{`...
